website/src/components/SearchPage/DownloadDialog/FieldSelector/FieldSelectorModal.tsx

```diff
@@ -0,0 +1,141 @@
+import { useState, type FC } from 'react';
+
+import { ACCESSION_VERSION_FIELD } from '../../../../settings.ts';
+import { type Metadata } from '../../../../types/config.ts';
+import { BaseDialog } from '../../../common/BaseDialog.tsx';
+
+type FieldSelectorProps = {
+    isOpen: boolean;
+    onClose: () => void;
+    metadata: Metadata[];
+    initialSelectedFields?: string[];
+    onSave: (selectedFields: string[]) => void;
+};
+
+export const FieldSelectorModal: FC<FieldSelectorProps> = ({
+    isOpen,
+    onClose,
+    metadata,
+    initialSelectedFields,
+    onSave,
+}) => {
+    // Ensure ACCESSION_VERSION_FIELD is always included in the selected fields
+    const getInitialSelectedFields = () => {
+        const fields = new Set(initialSelectedFields ?? getDefaultSelectedFields(metadata));
+        fields.add(ACCESSION_VERSION_FIELD);
+        return fields;
+    };
+
+    const [selectedFields, setSelectedFields] = useState<Set<string>>(getInitialSelectedFields());
+
+    const handleToggleField = (fieldName: string) => {
+        // Prevent unticking ACCESSION_VERSION_FIELD
+        if (fieldName === ACCESSION_VERSION_FIELD) {
+            return;
+        }
+
+        const newSelectedFields = new Set(selectedFields);
+        if (newSelectedFields.has(fieldName)) {
+            newSelectedFields.delete(fieldName);
+        } else {
+            newSelectedFields.add(fieldName);
+        }
+
+        // Ensure ACCESSION_VERSION_FIELD is always included
+        newSelectedFields.add(ACCESSION_VERSION_FIELD);
+
+        setSelectedFields(newSelectedFields);
+        // Apply changes immediately
+        onSave(Array.from(newSelectedFields));
+    };
+
+    // Group fields by header
+    const fieldsByHeader = metadata.reduce<Record<string, Metadata[]>>((acc, field) => {
+        const header = field.header ?? 'Other';
+        acc[header] = acc[header] ?? [];
+        acc[header].push(field);
+        return acc;
+    }, {});
+
+    // Sort headers alphabetically, but keep "Other" at the end
+    const sortedHeaders = Object.keys(fieldsByHeader).sort((a, b) => {
+        if (a === 'Other') return 1;
+        if (b === 'Other') return -1;
+        return a.localeCompare(b);
+    });
+
+    return (
+        <BaseDialog title='Select Fields to Download' isOpen={isOpen} onClose={onClose}>
+            <div className='mt-4 max-h-[60vh] overflow-y-auto p-2'>
+                {sortedHeaders.map((header) => (
+                    <div key={header} className='mb-6'>
+                        <h3 className='font-medium text-lg mb-2 text-gray-700'>{header}</h3>
+                        <div className='grid grid-cols-1 md:grid-cols-2 gap-x-4 gap-y-2'>
```

Comment:
Should we allow more columns on wider screens? On my screen, I can imagine 3-4 columns.

![image](https://github.com/user-attachments/assets/d9bca6d5-8cda-4d95-b1d7-04e539d7b933)

Reply:
Alternatively, it would make sense to set a max. width for the modal.

Reply:
👍 I've put a hacky max width in - if you prefer to do it in another way I'd prefer we make it a follow up issue